### ISSUE TITLE
UTF-8 parsing issue with MySQL native JSON columns

### ISSUE THREAD

**Issue by hellmund** [airbytehq/airbyte#10032]
## Environment
- **Airbyte version**:  0.35.0-alpha
- **OS Version / Instance**: Amazon Linux 2 AMI, EC2 t3a.large
- **Deployment**: Docker
- **Source Connector and version**: MySQL 0.5.1
- **Destination Connector and version**: Redshift 0.3.23
- **Severity**: High
- **Step where error happened**: Sync job

## Current Behavior
MySQL native JSON columns don't get parsed as UTF-8, resulting in broken encoding of non-ascii characters. Broken encoding is already present in _airbyte_raw tables. Standard VARCHAR columns with UTF-8 encoding get parsed correctly.

## Expected Behavior
MySQL native JSON fields containing UTF-8 encoded characters get parsed correctly - just like VARCHAR columns.

## Logs

<details>
<summary>Example data and Airbyte Log</summary> 

Example source data:

```

create schema airbyte_test collate utf8mb4_unicode_ci;

create table test_table_1
(
    id           int unsigned auto_increment
        primary key,
    test_varchar varchar(100)                        null,
    test_json    json                                null,
    updated_at   timestamp default CURRENT_TIMESTAMP not null on update CURRENT_TIMESTAMP
);

INSERT INTO airbyte_test.test_table_1 (id, test_varchar, test_json, updated_at) VALUES (1, 'Name1', null, '2022-02-02 21:13:12');
INSERT INTO airbyte_test.test_table_1 (id, test_varchar, test_json, updated_at) VALUES (2, 'Name2', '{"foo": "bar"}', '2022-02-02 21:13:12');
INSERT INTO airbyte_test.test_table_1 (id, test_varchar, test_json, updated_at) VALUES (3, 'Näme3', '{"foo": "bär"}', '2022-02-02 21:13:12');

```

Resulting Redshift destination data:

```

create table test_table_1
(
    id                           double precision,
    test_json                    varchar(65535),
    updated_at                   varchar(65535),
    test_varchar                 varchar(65535),
    _airbyte_ab_id               varchar(256) distkey,
    _airbyte_emitted_at          timestamp with time zone,
    _airbyte_normalized_at       timestamp encode az64,
    _airbyte_test_table_1_hashid varchar(32)
)
    sortkey (_airbyte_emitted_at);

alter table test_table_1
    owner to airbyte;

INSERT INTO stg_airbyte_test.test_table_1 (id, test_json, updated_at, test_varchar, _airbyte_ab_id, _airbyte_emitted_at, _airbyte_normalized_at, _airbyte_test_table_1_hashid) VALUES (1, null, '2022-02-02T21:13:12Z', 'Name1', '237f07e6-00f5-4a30-aa1a-31c4f362fd92', '2022-02-02 21:31:13.999000 +00:00', '2022-02-02 21:31:27.000000', '37468080924c4cff4fcad41f2a7e6270');
INSERT INTO stg_airbyte_test.test_table_1 (id, test_json, updated_at, test_varchar, _airbyte_ab_id, _airbyte_emitted_at, _airbyte_normalized_at, _airbyte_test_table_1_hashid) VALUES (2, '{\"foo\": \"bar\"}', '2022-02-02T21:13:12Z', 'Name2', 'd566e144-405f-4604-b33c-1555641e61f2', '2022-02-02 21:31:13.999000 +00:00', '2022-02-02 21:31:27.000000', 'ab6abba801c018c300a0d323346b3c5c');
INSERT INTO stg_airbyte_test.test_table_1 (id, test_json, updated_at, test_varchar, _airbyte_ab_id, _airbyte_emitted_at, _airbyte_normalized_at, _airbyte_test_table_1_hashid) VALUES (3, '{\"foo\": \"bÃ¤r\"}', '2022-02-02T21:13:12Z', 'Näme3', 'fc7d4d13-0e8e-4ad7-93b0-b654748aaa2e', '2022-02-02 21:31:13.999000 +00:00', '2022-02-02 21:31:27.000000', 'a221c89ec4cdce0e48cd2ef791b00436');

```

Airbyte log

```

2022-02-02 21:31:06 [32mINFO[m i.a.w.w.WorkerRun(call):49 - Executing worker wrapper. Airbyte version: 0.35.5-alpha
2022-02-02 21:31:06 [32mINFO[m i.a.w.t.TemporalAttemptExecution(get):118 - Docker volume job log path: /tmp/workspace/3883/0/logs.log
2022-02-02 21:31:06 [32mINFO[m i.a.w.t.TemporalAttemptExecution(get):123 - Executing worker wrapper. Airbyte version: version not set
2022-02-02 21:31:06 [33mWARN[m i.a.d.Databases(createPostgresDatabaseWithRetryTimeout):65 - Waiting for database to become available...
2022-02-02 21:31:06 [32mINFO[m i.a.d.i.j.JobsDatabaseInstance(lambda$static$2):25 - Testing if jobs database is ready...
2022-02-02 21:31:06 [32mINFO[m i.a.d.Databases(createPostgresDatabaseWithRetryTimeout):90 - Database available!
2022-02-02 21:31:06 [32mINFO[m i.a.d.Databases(createPostgresDatabaseWithRetry):48 - Database available!
2022-02-02 21:31:06 [32mINFO[m i.a.w.DefaultReplicationWorker(run):99 - start sync worker. job id: 3883 attempt id: 0
2022-02-02 21:31:06 [32mINFO[m i.a.w.DefaultReplicationWorker(run):108 - configured sync modes: {airbyte_test.test_table_1=full_refresh - overwrite}
2022-02-02 21:31:06 [32mINFO[m i.a.w.p.a.DefaultAirbyteDestination(start):69 - Running destination...
2022-02-02 21:31:06 [32mINFO[m i.a.c.i.LineGobbler(voidCall):82 - Checking if airbyte/destination-redshift:0.3.23 exists...
2022-02-02 21:31:07 [32mINFO[m i.a.c.i.LineGobbler(voidCall):82 - airbyte/destination-redshift:0.3.23 was found locally.
2022-02-02 21:31:07 [32mINFO[m i.a.w.p.DockerProcessFactory(create):177 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/3883/0 --network host --log-driver none airbyte/destination-redshift:0.3.23 write --config destination_config.json --catalog destination_catalog.json
2022-02-02 21:31:07 [32mINFO[m i.a.c.i.LineGobbler(voidCall):82 - Checking if airbyte/source-mysql:0.5.1 exists...
2022-02-02 21:31:07 [32mINFO[m i.a.c.i.LineGobbler(voidCall):82 - airbyte/source-mysql:0.5.1 was found locally.
2022-02-02 21:31:07 [32mINFO[m i.a.w.p.DockerProcessFactory(create):177 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/3883/0 --network host --log-driver none airbyte/source-mysql:0.5.1 read --config source_config.json --catalog source_catalog.json
2022-02-02 21:31:07 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getDestinationOutputRunnable$4):277 - Destination output thread started.
2022-02-02 21:31:07 [32mINFO[m i.a.w.DefaultReplicationWorker(run):130 - Waiting for source and destination threads to complete.
2022-02-02 21:31:07 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$3):238 - Replication thread started.
2022-02-02 21:31:08 [43mdestination[0m > SLF4J: Class path contains multiple SLF4J bindings.
2022-02-02 21:31:08 [43mdestination[0m > SLF4J: Found binding in [jar:file:/airbyte/lib/log4j-slf4j-impl-2.16.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-02-02 21:31:08 [43mdestination[0m > SLF4J: Found binding in [jar:file:/airbyte/lib/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-02-02 21:31:08 [43mdestination[0m > SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
2022-02-02 21:31:08 [44msource[0m > SLF4J: Class path contains multiple SLF4J bindings.
2022-02-02 21:31:08 [44msource[0m > SLF4J: Found binding in [jar:file:/airbyte/lib/log4j-slf4j-impl-2.15.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-02-02 21:31:08 [44msource[0m > SLF4J: Found binding in [jar:file:/airbyte/lib/slf4j-log4j12-1.7.30.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-02-02 21:31:08 [44msource[0m > SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
2022-02-02 21:31:09 [43mdestination[0m > SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
2022-02-02 21:31:09 [44msource[0m > SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
2022-02-02 21:31:11 [43mdestination[0m > 2022-02-02 21:31:11 [32mINFO[m i.a.i.d.r.RedshiftDestination(main):77 - {} - starting destination: class io.airbyte.integrations.destination.redshift.RedshiftDestination
2022-02-02 21:31:11 [44msource[0m > 2022-02-02 21:31:11 [32mINFO[m i.a.i.s.m.MySqlSource(main):203 - {} - starting source: class io.airbyte.integrations.source.mysql.MySqlSource
2022-02-02 21:31:11 [43mdestination[0m > 2022-02-02 21:31:11 [32mINFO[m i.a.i.b.IntegrationRunner(run):76 - {} - Running integration: io.airbyte.integrations.destination.redshift.RedshiftDestination
2022-02-02 21:31:11 [43mdestination[0m > 2022-02-02 21:31:11 [32mINFO[m i.a.i.b.IntegrationCliParser(parseOptions):118 - {} - integration args: {catalog=destination_catalog.json, write=null, config=destination_config.json}
2022-02-02 21:31:11 [43mdestination[0m > 2022-02-02 21:31:11 [32mINFO[m i.a.i.b.IntegrationRunner(run):80 - {} - Command: WRITE
2022-02-02 21:31:11 [43mdestination[0m > 2022-02-02 21:31:11 [32mINFO[m i.a.i.b.IntegrationRunner(run):81 - {} - Integration config: IntegrationConfig{command=WRITE, configPath='destination_config.json', catalogPath='destination_catalog.json', statePath='null'}
2022-02-02 21:31:12 [44msource[0m > 2022-02-02 21:31:12 [32mINFO[m i.a.i.b.IntegrationRunner(run):76 - {} - Running integration: io.airbyte.integrations.base.ssh.SshWrappedSource
2022-02-02 21:31:12 [44msource[0m > 2022-02-02 21:31:12 [32mINFO[m i.a.i.b.IntegrationCliParser(parseOptions):118 - {} - integration args: {read=null, catalog=source_catalog.json, config=source_config.json}
2022-02-02 21:31:12 [44msource[0m > 2022-02-02 21:31:12 [32mINFO[m i.a.i.b.IntegrationRunner(run):80 - {} - Command: READ
2022-02-02 21:31:12 [44msource[0m > 2022-02-02 21:31:12 [32mINFO[m i.a.i.b.IntegrationRunner(run):81 - {} - Integration config: IntegrationConfig{command=READ, configPath='source_config.json', catalogPath='source_catalog.json', statePath='null'}
2022-02-02 21:31:12 [43mdestination[0m > 2022-02-02 21:31:12 [33mWARN[m c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword examples - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2022-02-02 21:31:12 [43mdestination[0m > 2022-02-02 21:31:12 [33mWARN[m c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword airbyte_secret - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2022-02-02 21:31:12 [43mdestination[0m > 2022-02-02 21:31:12 [32mINFO[m i.a.i.d.j.c.SwitchingDestination(getConsumer):65 - {} - Using destination type: COPY_S3
2022-02-02 21:31:13 [44msource[0m > 2022-02-02 21:31:13 [33mWARN[m c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword order - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2022-02-02 21:31:13 [44msource[0m > 2022-02-02 21:31:13 [33mWARN[m c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword examples - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2022-02-02 21:31:13 [44msource[0m > 2022-02-02 21:31:13 [33mWARN[m c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword airbyte_secret - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2022-02-02 21:31:13 [44msource[0m > 2022-02-02 21:31:13 [33mWARN[m c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword multiline - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2022-02-02 21:31:13 [44msource[0m > 2022-02-02 21:31:13 [32mINFO[m i.a.i.b.s.SshTunnel(getInstance):170 - {} - Starting connection with method: NO_TUNNEL
2022-02-02 21:31:13 [44msource[0m > 2022-02-02 21:31:13 [32mINFO[m i.a.i.s.r.CdcStateManager(<init>):26 - {} - Initialized CDC state with: null
2022-02-02 21:31:13 [44msource[0m > 2022-02-02 21:31:13 [32mINFO[m i.a.i.s.r.StateManager(createCursorInfoForStream):118 - {} - No cursor field set in catalog but not present in state. Stream: AirbyteStreamNameNamespacePair{name='test_table_1', namespace='airbyte_test'}, New Cursor Field: null. Resetting cursor value
2022-02-02 21:31:14 [43mdestination[0m > 2022-02-02 21:31:14 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(startTracked):125 - {} - class io.airbyte.integrations.destination.buffered_stream_consumer.BufferedStreamConsumer started.
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.j.AbstractJdbcSource(lambda$discoverInternal$5):136 - {} - Table test_table_1 column id (type INT UNSIGNED[10]) -> Json type NUMBER
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.j.AbstractJdbcSource(lambda$discoverInternal$5):136 - {} - Table test_table_1 column test_varchar (type VARCHAR[100]) -> Json type STRING
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.j.AbstractJdbcSource(lambda$discoverInternal$5):136 - {} - Table test_table_1 column test_json (type JSON[1073741824]) -> Json type STRING
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.j.AbstractJdbcSource(lambda$discoverInternal$5):136 - {} - Table test_table_1 column updated_at (type TIMESTAMP[19]) -> Json type STRING
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.m.MySqlSource(getIncrementalIterators):186 - {} - using CDC: false
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.r.AbstractRelationalDbSource(queryTableFullRefresh):35 - {} - Queueing query for table: test_table_1
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.r.AbstractDbSource(lambda$read$2):123 - {} - Closing database connection pool.
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.r.AbstractDbSource(lambda$read$2):125 - {} - Closed database connection pool.
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.b.IntegrationRunner(run):133 - {} - Completed integration: io.airbyte.integrations.base.ssh.SshWrappedSource
2022-02-02 21:31:15 [44msource[0m > 2022-02-02 21:31:15 [32mINFO[m i.a.i.s.m.MySqlSource(main):205 - {} - completed source: class io.airbyte.integrations.source.mysql.MySqlSource
2022-02-02 21:31:16 [32mINFO[m i.a.w.DefaultReplicationWorker(run):135 - One of source or destination thread complete. Waiting on the other.
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.b.FailureTrackingAirbyteMessageConsumer(close):60 - {} - Airbyte message consumer: succeeded.
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(close):200 - {} - executing on success close procedure.
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(prepareStagingFile):95 - {} - S3 upload part size: 10 MB
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.d.s.c.S3CsvWriter(<init>):56 - {} - Full S3 path for stream 'test_table_1': s3://MYS3BUCKETNAME/stg_airbyte_test/test_table_1/2022_02_02_1643837474905_2664da96-675e-44c1-8755-c119ff28ec5d.csv
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.d.s.u.S3StreamTransferManagerHelper(getDefault):55 - {} - PartSize arg is set to 10 MB
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m a.m.s.StreamTransferManager(getMultiPartOutputStreams):329 - {} - Initiated multipart upload to MYS3BUCKETNAME/stg_airbyte_test/test_table_1/2022_02_02_1643837474905_2664da96-675e-44c1-8755-c119ff28ec5d.csv with full ID 16r9ohlHOSOnIrDux7bXn8TxXBb5_Qx8Gh9JS45tkUtqC.r03QJvL_GU9VwLCeWCQ1.30NKd0Q.J0Z6ILz0nlQZYIaROAyyRAbq2X6NEpWB01bZbAFvGu3o.ssJKkmt4
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.d.s.w.BaseS3Writer(close):107 - {} - Uploading remaining data for stream 'test_table_1'.
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m a.m.s.MultiPartOutputStream(close):158 - {} - Called close() on [MultipartOutputStream for parts 1 - 10000]
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m a.m.s.MultiPartOutputStream(close):158 - {} - Called close() on [MultipartOutputStream for parts 1 - 10000]
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [33mWARN[m a.m.s.MultiPartOutputStream(close):160 - {} - [MultipartOutputStream for parts 1 - 10000] is already closed
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m a.m.s.StreamTransferManager(complete):367 - {} - [Manager uploading to MYS3BUCKETNAME/stg_airbyte_test/test_table_1/2022_02_02_1643837474905_2664da96-675e-44c1-8755-c119ff28ec5d.csv with id 16r9ohlHO....ssJKkmt4]: Uploading leftover stream [Part number 1 containing 0.00 MB]
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m a.m.s.StreamTransferManager(uploadStreamPart):558 - {} - [Manager uploading to MYS3BUCKETNAME/stg_airbyte_test/test_table_1/2022_02_02_1643837474905_2664da96-675e-44c1-8755-c119ff28ec5d.csv with id 16r9ohlHO....ssJKkmt4]: Finished uploading [Part number 1 containing 0.00 MB]
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m a.m.s.StreamTransferManager(complete):395 - {} - [Manager uploading to MYS3BUCKETNAME/stg_airbyte_test/test_table_1/2022_02_02_1643837474905_2664da96-675e-44c1-8755-c119ff28ec5d.csv with id 16r9ohlHO....ssJKkmt4]: Completed
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.d.s.w.BaseS3Writer(close):109 - {} - Upload completed for stream 'test_table_1'.
2022-02-02 21:31:16 [43mdestination[0m > 2022-02-02 21:31:16 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(createDestinationSchema):137 - {} - Creating schema in destination if it doesn't exist: stg_airbyte_test
2022-02-02 21:31:17 [43mdestination[0m > 2022-02-02 21:31:17 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(createTemporaryTable):143 - {} - Preparing tmp table in destination for stream: test_table_1, schema: stg_airbyte_test, tmp table name: _airbyte_tmp_dle_test_table_1.
2022-02-02 21:31:17 [43mdestination[0m > 2022-02-02 21:31:17 [32mINFO[m i.a.i.d.r.RedshiftStreamCopier(copyStagingFileToTemporaryTable):85 - {} - Starting copy to tmp table: _airbyte_tmp_dle_test_table_1 in destination for stream: test_table_1, schema: stg_airbyte_test, .
2022-02-02 21:31:18 [43mdestination[0m > 2022-02-02 21:31:18 [32mINFO[m i.a.i.d.r.RedshiftStreamCopier(copyStagingFileToTemporaryTable):89 - {} - Copy to tmp table _airbyte_tmp_dle_test_table_1 in destination for stream test_table_1 complete.
2022-02-02 21:31:18 [43mdestination[0m > 2022-02-02 21:31:18 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(createDestinationTable):160 - {} - Preparing table _airbyte_raw_test_table_1 in destination.
2022-02-02 21:31:18 [43mdestination[0m > 2022-02-02 21:31:18 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(createDestinationTable):162 - {} - Table _airbyte_tmp_dle_test_table_1 in destination prepared.
2022-02-02 21:31:18 [43mdestination[0m > 2022-02-02 21:31:18 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(generateMergeStatement):169 - {} - Preparing to merge tmp table _airbyte_tmp_dle_test_table_1 to dest table: _airbyte_raw_test_table_1, schema: stg_airbyte_test, in destination.
2022-02-02 21:31:18 [43mdestination[0m > 2022-02-02 21:31:18 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(generateMergeStatement):173 - {} - Destination OVERWRITE mode detected. Dest table: _airbyte_raw_test_table_1, schema: stg_airbyte_test, truncated.
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(removeFileAndDropTmpTable):186 - {} - Begin cleaning s3 staging file stg_airbyte_test/test_table_1/2022_02_02_1643837474905_2664da96-675e-44c1-8755-c119ff28ec5d.csv.
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(removeFileAndDropTmpTable):190 - {} - S3 staging file stg_airbyte_test/test_table_1/2022_02_02_1643837474905_2664da96-675e-44c1-8755-c119ff28ec5d.csv cleaned.
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(removeFileAndDropTmpTable):194 - {} - Begin cleaning _airbyte_tmp_dle_test_table_1 tmp table in destination.
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(removeFileAndDropTmpTable):196 - {} - _airbyte_tmp_dle_test_table_1 tmp table in destination cleaned.
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.d.r.RedshiftStreamCopier(removeFileAndDropTmpTable):105 - {} - Begin cleaning s3 manifest file null/61463809-ac63-4b00-8a61-083c8d4f86f3/stg_airbyte_test/555111fc-6343-47f6-87ce-79cdc40e330a.manifest.
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.d.r.RedshiftStreamCopier(removeFileAndDropTmpTable):109 - {} - S3 manifest file null/61463809-ac63-4b00-8a61-083c8d4f86f3/stg_airbyte_test/555111fc-6343-47f6-87ce-79cdc40e330a.manifest cleaned.
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.b.IntegrationRunner(run):133 - {} - Completed integration: io.airbyte.integrations.destination.redshift.RedshiftDestination
2022-02-02 21:31:19 [43mdestination[0m > 2022-02-02 21:31:19 [32mINFO[m i.a.i.d.r.RedshiftDestination(main):79 - {} - completed destination: class io.airbyte.integrations.destination.redshift.RedshiftDestination
2022-02-02 21:31:19 [32mINFO[m i.a.w.DefaultReplicationWorker(run):137 - Source and destination threads complete.
2022-02-02 21:31:19 [32mINFO[m i.a.w.DefaultReplicationWorker(run):200 - sync summary: io.airbyte.config.ReplicationAttemptSummary@3e618946[status=completed,recordsSynced=3,bytesSynced=271,startTime=1643837466977,endTime=1643837479776,totalStats=io.airbyte.config.SyncStats@2ab55304[recordsEmitted=3,bytesEmitted=271,stateMessagesEmitted=0,recordsCommitted=3],streamStats=[io.airbyte.config.StreamSyncStats@65980fef[streamName=test_table_1,stats=io.airbyte.config.SyncStats@1f544d7b[recordsEmitted=3,bytesEmitted=271,stateMessagesEmitted=<null>,recordsCommitted=3]]]]
2022-02-02 21:31:19 [32mINFO[m i.a.w.DefaultReplicationWorker(run):209 - Source did not output any state messages
2022-02-02 21:31:19 [33mWARN[m i.a.w.DefaultReplicationWorker(run):220 - State capture: No state retained.
2022-02-02 21:31:19 [32mINFO[m i.a.w.t.TemporalAttemptExecution(get):144 - Stopping cancellation check scheduling...
2022-02-02 21:31:19 [32mINFO[m i.a.w.t.s.ReplicationActivityImpl(replicate):144 - sync summary: io.airbyte.config.StandardSyncOutput@13c652de[standardSyncSummary=io.airbyte.config.StandardSyncSummary@342e1300[status=completed,recordsSynced=3,bytesSynced=271,startTime=1643837466977,endTime=1643837479776,totalStats=io.airbyte.config.SyncStats@2ab55304[recordsEmitted=3,bytesEmitted=271,stateMessagesEmitted=0,recordsCommitted=3],streamStats=[io.airbyte.config.StreamSyncStats@65980fef[streamName=test_table_1,stats=io.airbyte.config.SyncStats@1f544d7b[recordsEmitted=3,bytesEmitted=271,stateMessagesEmitted=<null>,recordsCommitted=3]]]],state=<null>,outputCatalog=io.airbyte.protocol.models.ConfiguredAirbyteCatalog@7478db9f[streams=[io.airbyte.protocol.models.ConfiguredAirbyteStream@3679491a[stream=io.airbyte.protocol.models.AirbyteStream@1788ada9[name=test_table_1,jsonSchema={"type":"object","properties":{"id":{"type":"number"},"test_json":{"type":"string"},"updated_at":{"type":"string"},"test_varchar":{"type":"string"}}},supportedSyncModes=[full_refresh, incremental],sourceDefinedCursor=<null>,defaultCursorField=[],sourceDefinedPrimaryKey=[[id]],namespace=stg_airbyte_test,additionalProperties={}],syncMode=full_refresh,cursorField=[],destinationSyncMode=overwrite,primaryKey=[[id]],additionalProperties={}]],additionalProperties={}]]
2022-02-02 21:31:19 [32mINFO[m i.a.w.t.TemporalAttemptExecution(get):118 - Docker volume job log path: /tmp/workspace/3883/0/logs.log
2022-02-02 21:31:19 [32mINFO[m i.a.w.t.TemporalAttemptExecution(get):123 - Executing worker wrapper. Airbyte version: version not set
2022-02-02 21:31:19 [33mWARN[m i.a.d.Databases(createPostgresDatabaseWithRetryTimeout):65 - Waiting for database to become available...
2022-02-02 21:31:19 [32mINFO[m i.a.d.i.j.JobsDatabaseInstance(lambda$static$2):25 - Testing if jobs database is ready...
2022-02-02 21:31:19 [32mINFO[m i.a.d.Databases(createPostgresDatabaseWithRetryTimeout):90 - Database available!
2022-02-02 21:31:19 [32mINFO[m i.a.d.Databases(createPostgresDatabaseWithRetry):48 - Database available!
2022-02-02 21:31:19 [32mINFO[m i.a.w.DefaultNormalizationWorker(run):46 - Running normalization.
2022-02-02 21:31:19 [32mINFO[m i.a.w.n.DefaultNormalizationRunner(runProcess):122 - Running with normalization version: airbyte/normalization:0.1.63
2022-02-02 21:31:19 [32mINFO[m i.a.c.i.LineGobbler(voidCall):82 - Checking if airbyte/normalization:0.1.63 exists...
2022-02-02 21:31:19 [32mINFO[m i.a.c.i.LineGobbler(voidCall):82 - airbyte/normalization:0.1.63 was found locally.
2022-02-02 21:31:19 [32mINFO[m i.a.w.p.DockerProcessFactory(create):177 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/3883/0/normalize --network host --log-driver none airbyte/normalization:0.1.63 run --integration-type redshift --config destination_config.json --catalog destination_catalog.json
2022-02-02 21:31:20 [42mnormalization[0m > Running: transform-config --config destination_config.json --integration-type redshift --out /data/3883/0/normalize
2022-02-02 21:31:20 [42mnormalization[0m > Namespace(config='destination_config.json', integration_type=<DestinationType.redshift: 'redshift'>, out='/data/3883/0/normalize')
2022-02-02 21:31:20 [42mnormalization[0m > transform_redshift
2022-02-02 21:31:20 [42mnormalization[0m > Running: transform-catalog --integration-type redshift --profile-config-dir /data/3883/0/normalize --catalog destination_catalog.json --out /data/3883/0/normalize/models/generated/ --json-column _airbyte_data
2022-02-02 21:31:21 [42mnormalization[0m > Processing destination_catalog.json...
2022-02-02 21:31:21 [42mnormalization[0m >   Generating airbyte_ctes/stg_airbyte_test/test_table_1_ab1.sql from test_table_1
2022-02-02 21:31:21 [42mnormalization[0m >   Generating airbyte_ctes/stg_airbyte_test/test_table_1_ab2.sql from test_table_1
2022-02-02 21:31:21 [42mnormalization[0m >   Generating airbyte_ctes/stg_airbyte_test/test_table_1_ab3.sql from test_table_1
2022-02-02 21:31:21 [42mnormalization[0m >   Generating airbyte_tables/stg_airbyte_test/test_table_1.sql from test_table_1
2022-02-02 21:31:21 [42mnormalization[0m > detected no config file for ssh, assuming ssh is off.
2022-02-02 21:31:24 [42mnormalization[0m > Running with dbt=0.21.1
2022-02-02 21:31:24 [42mnormalization[0m > Unable to do partial parsing because ../build/partial_parse.msgpack not found
2022-02-02 21:31:26 [42mnormalization[0m > [[33mWARNING[0m]: Configuration paths exist in your dbt_project.yml file which do not apply to any resources.
2022-02-02 21:31:26 [42mnormalization[0m > There are 2 unused configuration paths:
2022-02-02 21:31:26 [42mnormalization[0m > - models.airbyte_utils.generated.airbyte_views
2022-02-02 21:31:26 [42mnormalization[0m > - models.airbyte_utils.generated.airbyte_incremental
2022-02-02 21:31:26 [42mnormalization[0m > 
2022-02-02 21:31:26 [42mnormalization[0m > Found 4 models, 0 tests, 0 snapshots, 0 analyses, 520 macros, 0 operations, 0 seed files, 1 source, 0 exposures
2022-02-02 21:31:26 [42mnormalization[0m > 
2022-02-02 21:31:27 [42mnormalization[0m > 21:31:27 | Concurrency: 32 threads (target='prod')
2022-02-02 21:31:27 [42mnormalization[0m > 21:31:27 | 
2022-02-02 21:31:27 [42mnormalization[0m > 21:31:27 | 1 of 1 START table model stg_airbyte_test.test_table_1....................................................... [RUN]
2022-02-02 21:31:28 [42mnormalization[0m > 21:31:28 | 1 of 1 OK created table model stg_airbyte_test.test_table_1.................................................. [[32mSELECT[0m in 1.23s]
2022-02-02 21:31:28 [42mnormalization[0m > 21:31:28 | 
2022-02-02 21:31:28 [42mnormalization[0m > 21:31:28 | Finished running 1 table model in 1.82s.
2022-02-02 21:31:28 [42mnormalization[0m > 
2022-02-02 21:31:28 [42mnormalization[0m > [32mCompleted successfully[0m
2022-02-02 21:31:28 [42mnormalization[0m > 
2022-02-02 21:31:28 [42mnormalization[0m > Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
2022-02-02 21:31:28 [32mINFO[m i.a.w.DefaultNormalizationWorker(run):69 - Normalization executed in 9 seconds.
2022-02-02 21:31:28 [32mINFO[m i.a.w.t.TemporalAttemptExecution(get):144 - Stopping cancellation check scheduling...

```

</details>

## Steps to Reproduce
1. have MySQL database with Unicode collation, e.g. utf8mb4_unicode_ci
2. have table with json column containing UTF-8 characters
3. sync with Airbyte to Redshift

## Are you willing to submit a PR?
No, but happy to verify fix with new builds.

**Comment by alafanechere** [comment]
Thank you @hellmund for the detailed explanation.

@tuliren another encoding problem 😄  Are you aware of this problem? Do you think it relates to an existing epic?

**Comment by marcosmarxm** [comment]
Zendesk ticket #[1491](https://airbyte7538.zendesk.com/agent/tickets/1491) has been linked to this issue.

**Comment by marcosmarxm** [comment]
*Comment made from Zendesk by Nataly on 2022-07-07 at 11:41:*

<blockquote><div class="zd-comment" dir="auto">Hi, @Nakachi! There's an open issue related to decoding non-ascii characters <a rel="noopener noreferrer" href="https://github.com/airbytehq/airbyte/issues/10032">here</a>. I've linked your post to it and will follow up with the database team to make sure it is on our roadmap!<br></div></blockquote>

**Comment by grishick** [comment]
> 

I filed a separate issue for Salesforce connector: https://github.com/airbytehq/airbyte/issues/14659

**Comment by tuliren** [comment]
I added a test case in #14865, and could not replicate this issue.

For context:
https://dev.mysql.com/doc/refman/8.0/en/json.html

> MySQL handles strings used in JSON context using the `utf8mb4` character set and `utf8mb4_bin` collation. Strings in other character sets are converted to `utf8mb4` as necessary. (For strings in the ascii or `utf8` character sets, no conversion is needed because ascii and `utf8` are subsets of `utf8mb4`.)

Looks like the character set or collation should not affect when the JSON data is parsed.
In addition, the code reads the JSON column by JDBC's `getString` method. There is no way to specify the encoding.

I will test this with an actual database as well.

**Comment by tuliren** [comment]
@hellmund, it looks like this issue has been resolved in the latest version of MySQL source and Redshift destination. Please see the comment above for details. Can you confirm that it is no longer an issue for you?

**Comment by data-evs** [comment]
Hi. I also got the same issue with Salesforce and BigQuery : https://discuss.airbyte.io/t/salesforce-bigquery-encoding-problem/1790/5
